html/modules/custom/reliefweb_import/src/Command/ReliefwebImportCommand.php

```diff
@@ -0,0 +1,680 @@
+          '@path' => $violation->getPropertyPath(),
+          '@message' => $violation->getMessage()->__toString(),
+          '@guid' => $job->field_job_id->value,
+        ]);
+      }
+
+      throw new ReliefwebImportExceptionSoftViolation(implode("\n", $errors));
+    }
+  }
+
+  /**
+   * Our logger.
+   */
+  protected function logger() {
+    return $this->loggerFactory->get('reliefweb_import');
+  }
+
+  /**
+   * Validate base URL.
+   */
+  protected function validateBaseUrl($base_url) {
+    if (empty(trim($base_url))) {
+      throw new ReliefwebImportExceptionViolation('Base URL is empty.');
+    }
+
+    if (!UrlHelper::isValid($base_url, TRUE)) {
+      throw new ReliefwebImportExceptionViolation('Base URL is not a valid link.');
+    }
+  }
+
+  /**
+   * Validate link.
+   */
+  protected function validateLink($link, $base_url) {
+    if (empty(trim($link))) {
+      throw new ReliefwebImportExceptionViolation('Feed item found without a link.');
+    }
+
+    if (!UrlHelper::isValid($link, TRUE)) {
+      throw new ReliefwebImportExceptionViolation('Invalid feed item link.');
+    }
+
+    if (strpos($link, $base_url) !== 0) {
+      throw new ReliefwebImportExceptionViolation('Invalid feed item link base.');
+    }
+  }
+
+  /**
+   * Validate title.
+   */
+  protected function validateTitle($title) {
+    if (empty(trim($title))) {
+      throw new ReliefwebImportExceptionViolation('Job found with empty title.');
+    }
+
+    // Clean the title.
+    $title = strip_tags($title);
+
+    $options = [
+      'line_breaks' => TRUE,
+      'consecutive' => TRUE,
+    ];
+    $title = $this->cleanText($title, $options);
+
+    // Ensure the title size is reasonable. The max length matches the one from
+    // the job form.
+    $length = mb_strlen($title);
+    if ($length < 10 || $length > 150) {
+      throw new ReliefwebImportExceptionViolation('Invalid title length.');
+    }
+
+    return $title;
+  }
+
+  /**
+   * Validate the body field of a feed item, clean and check its size.
+   *
+   * @param string $data
+   *   Raw data from XML.
+   */
+  protected function validateBody($data) {
+    // Clean the body field.
+    $body = $this->sanitizeText('body', $data, 'plain_text');
+
+    // Ensure the body field size is reasonable.
+    $length = mb_strlen($body);
+    if ($length < 400 || $length > 50000) {
+      throw new ReliefwebImportException(strtr('Invalid field size for body, @length characters found, has to be between 400 and 50000', [
+        '@length' => $length,
+      ]));
+    }
+
+    return $body;
+  }
+
+  /**
+   * Sanitize text, converting it to markdown.
+   *
+   * @param string $field
+   *   Field name.
+   * @param string $text
+   *   Field text content.
+   * @param string $format
+   *   Format to which conver the text if not already HTML.
+   *
+   * @return string
+   *   Sanitized content.
+   */
+  protected function sanitizeText($field, $text, $format = 'plain_text') {
```

Comment:
"plain_text" should be changed to "markdown". D7 "markdown" format was the "plain_text" one. In D9 plain_text is really plain text and "markdown" is the real markdown. That may be the reason of the mess up in the escaping when running check_markup.